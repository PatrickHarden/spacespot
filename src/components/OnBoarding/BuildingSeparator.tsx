import React from 'react'
import { useIntl } from 'react-intl'
import Building from 'components/icons/Building'
import IconSeparator from './IconSeparator'

const BuildingSeparator = () => {
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  return (
    <IconSeparator
      label={t('ONBOARDING_BUILDING_AND_AMMENITIES')}
      icon={<Building />}
    />
  )
}

export default BuildingSeparator
