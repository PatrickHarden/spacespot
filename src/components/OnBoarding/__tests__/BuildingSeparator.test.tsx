import React from 'react'
import BuildingSeparator from '../BuildingSeparator'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

jest.mock('react-intl', () => ({
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

it('renders without crashing', async () => {
  const wrapper = shallow(<BuildingSeparator />)
  expect(shallowToJson(wrapper)).toMatchSnapshot()
})
