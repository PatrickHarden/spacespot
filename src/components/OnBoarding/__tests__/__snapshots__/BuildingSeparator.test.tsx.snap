// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<IconSeparator
  icon={<Building />}
  label="ONBOARDING_BUILDING_AND_AMMENITIES"
/>
`;
